dingo/core/samplers.py

```diff
@@ -492,10 +492,19 @@ def _run_sampler(
 
             time_sample_start = time.time()
             self.model.model.eval()
+
             with torch.no_grad():
-                y, log_prob = self.model.model.sample_and_log_prob(
-                    x["data"], x["context_parameters"]
-                )
+                # If we are running with one detector, we need not pass extra context to
+                # the network. This is because there is a global shift which is applied to the
+                # geocent_time according to the time proxies.
+                # Here len(x["extrinsic_parameters"]) == 2, is only the 1 detector proxy and
+                # the geocent time.
```

Comment:
`core.samplers` should not have to know anything about gravitational waves. Maybe there is another way to do this...

Reply:
So the worry is that `x["context_parameters"]` is empty?